src/cluster.c

```diff
@@ -4656,17 +4669,35 @@ void migrateCommand(redisClient *c) {
     {
         char buf1[1024];
         char buf2[1024];
+        char buf3[1024];
 
-        /* Read the two replies */
-        if (select && syncReadLine(cs->fd, buf1, sizeof(buf1), timeout) <= 0)
+        /* Read the three replies */
+        if (auth && syncReadLine(cs->fd, buf1, sizeof(buf1), timeout) <= 0)
             goto socket_rd_err;
-        if (syncReadLine(cs->fd, buf2, sizeof(buf2), timeout) <= 0)
+        if (select && syncReadLine(cs->fd, buf2, sizeof(buf2), timeout) <= 0)
             goto socket_rd_err;
-        if ((select && buf1[0] == '-') || buf2[0] == '-') {
+        if (syncReadLine(cs->fd, buf3, sizeof(buf3), timeout) <= 0)
+            goto socket_rd_err;
+
+        // Do not need to return error about server password.
+        //
+        // If the server (c->argv[1]:c->argv[2]) have not set the password while the client
+        // sent its command with password, the following operations will continue to sucess.
+        //
+        // if the server have set the password while the client sent its command without any
+        // any password, the following operations will also fail and return the alert
+        // "NOAUTH Authentication required".
+        //
+        // if ((auth && buf1[0] == '-') || (select && buf2[0] == '-') || buf3[0] == '-') {
+        //     addReplyErrorFormat(c,"Target instance replied with error: %s",
+        //         (auth && buf1[0] == '-') ? buf1+1 : ((select && buf2[0] == '-') ? buf2+1 : buf3+1));
+        //     goto socket_rd_err;
+        // }
```

Comment:
If this is to be merged, we can reduce the comment to a small summary, no need to retain old code here.